src/renderer/components/outputs/GenericOutput.tsx

```diff
@@ -1,62 +1,74 @@
 import { Center, Flex, Spacer, Text } from '@chakra-ui/react';
-import { memo } from 'react';
-import { useContextSelector } from 'use-context-selector';
-import { OutputId } from '../../../common/common-types';
-import { Type } from '../../../common/types/types';
-import { GlobalVolatileContext } from '../../contexts/GlobalNodeState';
+import { memo, useEffect } from 'react';
+import { useContext, useContextSelector } from 'use-context-selector';
+import { NamedExpression, NamedExpressionField } from '../../../common/types/expression';
+import { StringLiteralType } from '../../../common/types/types';
+import { isStartingNode } from '../../../common/util';
+import { GlobalContext, GlobalVolatileContext } from '../../contexts/GlobalNodeState';
 import { TypeTag } from '../TypeTag';
-import { OutputContainer } from './OutputContainer';
-
-interface GenericOutputProps {
-    id: string;
-    label: string;
-    outputId: OutputId;
-    definitionType: Type;
-    useOutputData: (outputId: OutputId) => unknown;
-}
+import { OutputProps } from './props';
 
 export const GenericOutput = memo(
-    ({ label, id, outputId, definitionType, useOutputData }: GenericOutputProps) => {
+    ({ label, id, outputId, schemaId, useOutputData, kind }: OutputProps) => {
         const type = useContextSelector(GlobalVolatileContext, (c) =>
             c.typeState.functions.get(id)?.outputs.get(outputId)
         );
 
-        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        const { setManualOutputType, schemata } = useContext(GlobalContext);
+
+        const schema = schemata.get(schemaId);
+
         const value = useOutputData(outputId);
 
+        useEffect(() => {
+            if (isStartingNode(schema)) {
+                if (value) {
```

Comment:
The value might an empty string.

```suggestion
                if (value !== undefined) {
```